src/modules/settings-info/dto/create-settings-info.dto.ts

```diff
@@ -0,0 +1,19 @@
+import { IsEmail, IsPhoneNumber, IsString, MaxLength } from 'class-validator';
+
+export class CreateSettingsInfoDto {
+  @IsEmail()
+  @IsString()
+  readonly email: string;
```

Comment:
any of them could be empty, should not you setup rule for it?

Reply:
Yes, I'll change it. The only parameter that I believe should be mandatory is the email when sending a POST request in the DTO. If the email is not present the request will create a new user, which is the responsibility of the auth module.